datalad/distribution/create_sibling.py

```diff
@@ -146,46 +148,86 @@ class CreateSibling(Interface):
     @datasetmethod(name='create_sibling')
     def __call__(sshurl, name=None, target_dir=None,
                  target_url=None, target_pushurl=None,
-                 dataset=None, recursive=False,
+                 dataset=None,
+                 recursive=False,
+                 recursion_limit=None,
                  existing='error', shared=False, ui=False,
                  as_common_datasrc=None,
                  publish_by_default=None,
                  publish_depends=None):
 
-        if sshurl is None:
-            raise ValueError("""insufficient information for target creation
-            (needs at least a dataset and a SSH URL).""")
+        # there is no point in doing anything further
+        not_supported_on_windows(
+            "Support for SSH connections is not yet implemented in Windows")
 
-        if name is None and \
-                (target_url is not None or
-                 target_pushurl is not None):
-            raise ValueError("""insufficient information for adding the target
-            as a sibling (needs at least a name)""")
+        if not sshurl:
+            raise InsufficientArgumentsError(
+                "need at least an SSH URL")
 
-        # shortcut
-        ds = require_dataset(dataset, check_installed=True,
-                             purpose='creating a sibling')
-
-        assert(ds is not None and sshurl is not None and ds.repo is not None)
-
-        # determine target parameters:
+        # check the login URL
         sshri = RI(sshurl)
-
         if not isinstance(sshri, SSHRI) \
                 and not (isinstance(sshri, URL) and sshri.scheme == 'ssh'):
```

Comment:
:-) I used it with the entire connection sharing mechanic.
Therefore "should" means: "If I didn't miss an important point".